scripts/db_migrator.py

```diff
@@ -96,6 +100,89 @@ def migrate_interface_table(self):
+        from version with old SDK will be broken without migration.
+        This migration is specifically for Mellanox platform. 
+        """
+        # Buffer pools defined in version 1_0_2
+        buffer_pools = ['ingress_lossless_pool', 'egress_lossless_pool', 'ingress_lossy_pool', 'egress_lossy_pool']
+
+        # Old default buffer pool values on Mellanox platform 
+        spc1_t0_default_value = [{'ingress_lossless_pool': '4194304'}, {'egress_lossless_pool': '16777152'}, {'ingress_lossy_pool': '7340032'}, {'egress_lossy_pool': '7340032'}]
+        spc1_t1_default_value = [{'ingress_lossless_pool': '2097152'}, {'egress_lossless_pool': '16777152'}, {'ingress_lossy_pool': '5242880'}, {'egress_lossy_pool': '5242880'}]
+        spc2_t0_default_value = [{'ingress_lossless_pool': '8224768'}, {'egress_lossless_pool': '35966016'}, {'ingress_lossy_pool': '8224768'}, {'egress_lossy_pool': '8224768'}]
+        spc2_t1_default_value = [{'ingress_lossless_pool': '12042240'}, {'egress_lossless_pool': '35966016'}, {'ingress_lossy_pool': '12042240'}, {'egress_lossy_pool': '12042240'}]
+
+        # Get platform info and hwsku from config_db.json
+        config_db_file_path = "/etc/sonic/config_db.json"
+        if os.path.exists(config_db_file_path):
+            try:
+                with open(config_db_file_path) as config_db_file:
+                    config_db = json.load(config_db_file)
+                    device_data = config_db['DEVICE_METADATA']
+                    hwsku = device_data['localhost']['hwsku']
+                    platform = device_data['localhost']['platform']
+                    buffer_pool_conf = config_db['BUFFER_POOL']
+            except IOError:
+                log_error("failed to open config_db.json file, skip migration")
+                return False
+        else:
+            # Try to get related info from DB(in warm reboot case DB will be retored)
+            buffer_pool_conf = self.configDB.get_table('BUFFER_POOL')
+            device_data = self.configDB.get_table('DEVICE_METADATA')
+            if 'localhost' in device_data.keys():
+                hwsku = device_data['localhost']['hwsku']
+                platform = device_data['localhost']['platform']
+            else:
+                log_error("Trying to get data from DB but it's empty, skip migration")
+                return False
+
+        # Get current buffer pool configuration, only migrate configration which 
+        # with default values, if it's not default, leave it as is.
+        pool_size_in_db_list = []
+        pools_in_db = buffer_pool_conf.keys()
+
+        # Buffer pool numbers is different with default, don't need migrate
+        if len(pools_in_db) != len(buffer_pools):
+            return True
+
+        # If some buffer pool is not default ones, don't need migrate
+        for buffer_pool in buffer_pools:
+            if buffer_pool not in pools_in_db:
+                return True
```

Comment:
Should this be continue instead of return? Do you need to migrate other buffer pool(s) that might still have default values?

Reply:
the idea is that if the user changed the configuration,  and here we migrate it arbitrarily, then we are changing the user's configuration w/o notice, IMO the proper way here should elaborate the issue in release notes(we are going to add this issue to 201911 version release notes when we have a GA version) about this issue, and give a guideline to the user for the handle.